OSDOCS#13454: Updates to the web console tutorial (#91212)
openshift/openshift-docs: modules/getting-started-web-console-creating-secret.adoc

// Module included in the following assemblies:
//
// * tutorials/dev-app-web-console.adoc

:_mod-docs-content-type: PROCEDURE
[id="getting-started-web-console-creating-secret_{context}"]
= Providing access to the database by creating a secret

> The `nationalparks` application needs information, such as the database name, username, and passwords, on how to access the MongoDB database. However, because this information is sensitive, you should not store it directly in the pod.

Reviewer: ```suggestion
The `nationalparks` application needs information, such as the database name, username, and passwords, to access the MongoDB database. However, because this information is sensitive, you should not store it directly in the pod.
```

You can use a _secret_ to store sensitive information, and share that secret with workloads.

> `Secret` objects provide a mechanism to hold sensitive information such as passwords, {product-title} client configuration files, private source repository credentials, and so on. Secrets decouple sensitive content from the pods. You can mount secrets into containers by using a volume plugin, or the system can use secrets to perform actions on behalf of a pod.

Reviewer: ```suggestion
`Secret` objects provide a mechanism to hold sensitive information such as passwords, {product-title} client configuration files, and private source repository credentials. Secrets decouple sensitive content from the pods. You can mount secrets into containers by using a volume plugin. The system can then use secrets to perform actions on behalf of a pod.
```
Suggested this edit to the last sentence because they seem like two separate things that are sequential...? But I'm not very knowledgeable about this, so ignore if n/a.

Author: Good question. This was already here, but I agree that this doesn't seem to make total sense. It looks like it was taken directly from the "Understanding secrets" section here: https://91212--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-secrets#nodes-pods-secrets-about_nodes-pods-secrets

@AlexonOliveiraRH Any thoughts on how we could improve this wording? Should it be saying that you can either mount the secret by using the volume plugin or pass it as env vars (like we're doing here)? And then I'm not sure whether that last sentence ("The system can then use secrets to perform actions on behalf of a pod.") makes sense afterward? wdyt?

```suggestion
`Secret` objects provide a mechanism to hold sensitive information such as passwords, {product-title} client configuration files, and private source repository credentials. Secrets decouple sensitive content from the pods. You can mount secrets into containers by using a volume plugin or by passing the secret in as an environment variable. The system can then use secrets to perform actions on behalf of a pod.
```

Author: @AlexonOliveiraRH do you mean the way it originally was was correct?

> You can mount secrets into containers by using a volume plugin, or the system can use secrets to perform actions on behalf of a pod.

Or the way I've updated it to be now is correct?

> You can mount secrets into containers by using a volume plugin or by passing the secret in as an environment variable. The system can then use secrets to perform actions on behalf of a pod.

The following procedure creates the `nationalparks-mongodb-parameters` secret and mounts it to the `nationalparks` workload.

.Prerequisites

* You have deployed the `nationalparks` back-end application.
* You have deployed the `mongodb-nationalparks` database application.

.Procedure

. Navigate to *Workloads* -> *Secrets*.
. Click *Create* -> *Key/value secret*.
. In the *Secret name* field, enter `nationalparks-mongodb-parameters`.
.. Enter the following values for *Key* and *Value*:
+
.Secret keys and values
[cols="1,1"]
|===
|Key |Value

|`DATABASE_SERVICE_NAME`|`mongodb-nationalparks`
|`MONGODB_USER`|`mongodb`
|`MONGODB_PASSWORD`|`mongodb`
|`MONGODB_DATABASE`|`mongodb`
|`MONGODB_ADMIN_PASSWORD`|`mongodb`
|===
+
[TIP]
====
Click *Add key/value* to add each additional key/value pair.
====

. Click *Create*.
. Click *Add Secret to workload*.

> . From the *Add this secret to workload* drop-down list, select `nationalparks`.

Reviewer: ```suggestion
. From the *Add this secret to workload* list, select `nationalparks`.
```
